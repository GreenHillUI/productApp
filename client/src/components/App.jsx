import { lazy, React } from 'react';

const Overview = lazy(() => import('./overview/Overview'));
const QuestionList = lazy(() => import('./QuestionList'));
const RelatedProducts = lazy(() => import('./RelatedProducts'));
const Ratings = lazy(() => import('./Ratings'));
const Setter = lazy(() => import('./Setter'));
const Navbar = lazy(() => import('./navbar/Navbar'));

//NOTE: Should only have containers rendered inside, using QuestionList
//component temporarily, will change either when done with CSS or when container is built

function App() {
  return (
    <>
      <Navbar />
      <Setter />
      <Overview />
      <RelatedProducts />
      <QuestionList />
      <Ratings />
    </>
  );
}

export default App;
