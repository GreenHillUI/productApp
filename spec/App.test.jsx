/**
 * @jest-environment jsdom
 */
/* eslint-disable no-undef */
/* eslint-disable import/no-extraneous-dependencies */
import { Provider } from 'react-redux';
import App from '../client/src/components/App';
import store from '../client/src/store';

const React = require('react');
const { create } = require('react-test-renderer');
// const { render, screen } = require('@testing-library/react');

// const screen = require('@testing-library/react').screen;


describe('App Component Snapshot test', () => {

  test('adding or verifying App Snapshot', () => {

    const AppSnap = create(
      <Provider store={store}>
        <App />
      </Provider>
    );
    expect(AppSnap.toJSON()).toMatchSnapshot();
  });

});

