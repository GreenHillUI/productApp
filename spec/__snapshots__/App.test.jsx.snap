// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Snapshot test adding or verifying App Snapshot 1`] = `
Array [
  <div
    id="overview"
  >
    <div
      id="overviewTop"
    >
      <div
        id="overviewStars"
      >
        Review Score:
        <div
          className="stars"
        >
          <div
            className="empty-stars"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M480 208H308L256 48l-52 160H32l140 96-54 160 138-100 138 100-54-160z"
                fill="none"
                strokeLinejoin="round"
                strokeWidth="32"
              />
            </svg>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M480 208H308L256 48l-52 160H32l140 96-54 160 138-100 138 100-54-160z"
                fill="none"
                strokeLinejoin="round"
                strokeWidth="32"
              />
            </svg>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M480 208H308L256 48l-52 160H32l140 96-54 160 138-100 138 100-54-160z"
                fill="none"
                strokeLinejoin="round"
                strokeWidth="32"
              />
            </svg>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M480 208H308L256 48l-52 160H32l140 96-54 160 138-100 138 100-54-160z"
                fill="none"
                strokeLinejoin="round"
                strokeWidth="32"
              />
            </svg>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M480 208H308L256 48l-52 160H32l140 96-54 160 138-100 138 100-54-160z"
                fill="none"
                strokeLinejoin="round"
                strokeWidth="32"
              />
            </svg>
          </div>
          <div
            className="star-rating"
            style={
              Object {
                "width": "NaN%",
              }
            }
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M496 203.3H312.36L256 32l-56.36 171.3H16l150.21 105.4-58.5 171.3L256 373.84 404.29 480l-58.61-171.3z"
              />
            </svg>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M496 203.3H312.36L256 32l-56.36 171.3H16l150.21 105.4-58.5 171.3L256 373.84 404.29 480l-58.61-171.3z"
              />
            </svg>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M496 203.3H312.36L256 32l-56.36 171.3H16l150.21 105.4-58.5 171.3L256 373.84 404.29 480l-58.61-171.3z"
              />
            </svg>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M496 203.3H312.36L256 32l-56.36 171.3H16l150.21 105.4-58.5 171.3L256 373.84 404.29 480l-58.61-171.3z"
              />
            </svg>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M496 203.3H312.36L256 32l-56.36 171.3H16l150.21 105.4-58.5 171.3L256 373.84 404.29 480l-58.61-171.3z"
              />
            </svg>
          </div>
        </div>
        <div
          id="overviewReadAll"
        >
            read all  0 reviews  
        </div>
      </div>
      <h2
        className="overviewProductCategory"
      />
      <h1
        className="overviewProductName"
      />
      <div
        className="overviewPrice"
      >
        $undefined
      </div>
      <div
        id="addToCartContainer"
      >
        <form
          action="/"
          className="addToCartForm"
        >
          <label
            htmlFor="sizeSelect"
          >
            Size: :
            <select
              id="sizeSelect"
              onChange={[Function]}
              required={true}
            >
              <option
                value=""
              >
                Select Size
              </option>
              <option
                value={null}
              >
                 Out of Stock
              </option>
            </select>
          </label>
          <label
            htmlFor="quantitySelect"
          >
            Qty: :
            <select
              id="quantitySelect"
            >
              <option>
                -
              </option>
            </select>
          </label>
          <br />
          <input
            id="addToCartButton"
            type="submit"
            value="Add To Cart"
          />
        </form>
      </div>
    </div>
    <div
      id="gallery"
    >
      <div
        id="gallerySlideView"
      />
      <div
        className="galleryCarousel"
        id="galleryCarousel"
      >
        <div
          className="galleryThumbView"
        >
          <ul />
        </div>
        <button
          className="arrowButton"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="thumbArrow thumbArrow"
            fill="currentColor"
            height="1em"
            onClick={[Function]}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
                "fill": "black",
                "height": "1.5em",
                "width": "1.5em",
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
            <path
              d="M6.23 20.23L8 22l10-10L8 2 6.23 3.77 14.46 12z"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      id="overviewTextBlurb"
    >
      <div
        className="overviewProductSlogan"
      />
      <div
        className="overviewProductInfo"
      >
        <p
          className="overviewProductDescription"
        />
      </div>
    </div>
    <ul
      id="overviewFeatures"
    >
      Features:
      Loading
    </ul>
  </div>,
  <div
    id="related-items"
  >
    <h1>
      Related Products
    </h1>
    <ul
      className="product-carrousel"
    />
    <h1>
      Your Outfit
    </h1>
    <ul
      className="product-carrousel"
    />
  </div>,
  <div
    id="q-container"
  >
    <span
      id="q-title"
    >
      QUESTIONS & ANSWERS
    </span>
    <div
      id="q-nav"
    >
      <input
        id="q-search"
        onChange={[Function]}
        placeholder="HAVE A QUESTION? SEARCH FOR ANSWERS..."
        type="text"
      />
    </div>
    <ul
      id="q-list-col"
    >
      <h3>
        PLEASE WAIT... PRODUCTS LOADING
      </h3>
    </ul>
    <div
      id="q-buttons"
    >
      <button
        onClick={[Function]}
        type="button"
      >
         MORE ANSWERED QUESTIONS
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        ADD A QUESTION +
      </button>
    </div>
  </div>,
  <div
    id="ratings-reviews"
  >
    <div>
      Ratings And Reviews
    </div>
    <div
      className="review-summary"
    >
      <div
        id="big-review-num"
      >
        NaN
        <div
          className="stars"
        >
          <div
            className="empty-stars"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M480 208H308L256 48l-52 160H32l140 96-54 160 138-100 138 100-54-160z"
                fill="none"
                strokeLinejoin="round"
                strokeWidth="32"
              />
            </svg>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M480 208H308L256 48l-52 160H32l140 96-54 160 138-100 138 100-54-160z"
                fill="none"
                strokeLinejoin="round"
                strokeWidth="32"
              />
            </svg>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M480 208H308L256 48l-52 160H32l140 96-54 160 138-100 138 100-54-160z"
                fill="none"
                strokeLinejoin="round"
                strokeWidth="32"
              />
            </svg>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M480 208H308L256 48l-52 160H32l140 96-54 160 138-100 138 100-54-160z"
                fill="none"
                strokeLinejoin="round"
                strokeWidth="32"
              />
            </svg>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M480 208H308L256 48l-52 160H32l140 96-54 160 138-100 138 100-54-160z"
                fill="none"
                strokeLinejoin="round"
                strokeWidth="32"
              />
            </svg>
          </div>
          <div
            className="star-rating"
            style={
              Object {
                "width": "103%",
              }
            }
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M496 203.3H312.36L256 32l-56.36 171.3H16l150.21 105.4-58.5 171.3L256 373.84 404.29 480l-58.61-171.3z"
              />
            </svg>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M496 203.3H312.36L256 32l-56.36 171.3H16l150.21 105.4-58.5 171.3L256 373.84 404.29 480l-58.61-171.3z"
              />
            </svg>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M496 203.3H312.36L256 32l-56.36 171.3H16l150.21 105.4-58.5 171.3L256 373.84 404.29 480l-58.61-171.3z"
              />
            </svg>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M496 203.3H312.36L256 32l-56.36 171.3H16l150.21 105.4-58.5 171.3L256 373.84 404.29 480l-58.61-171.3z"
              />
            </svg>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M496 203.3H312.36L256 32l-56.36 171.3H16l150.21 105.4-58.5 171.3L256 373.84 404.29 480l-58.61-171.3z"
              />
            </svg>
          </div>
        </div>
      </div>
      <div>
        Loading...
      </div>
      <br />
      <div>
        <div
          onClick={[Function]}
        >
          <div>
            <div
              style={
                Object {
                  "position": "absolute",
                }
              }
            >
              5 stars
            </div>
            <div
              style={
                Object {
                  "background": "#D3D3D3",
                  "height": 5,
                  "left": 75,
                  "position": "absolute",
                  "width": 200,
                }
              }
            />
            <div
              style={
                Object {
                  "background": "green",
                  "height": 5,
                  "left": 75,
                  "position": "absolute",
                  "width": NaN,
                }
              }
            />
            <br />
          </div>
        </div>
        <br />
        <div
          onClick={[Function]}
        >
          <div>
            <div
              style={
                Object {
                  "position": "absolute",
                }
              }
            >
              4 stars
            </div>
            <div
              style={
                Object {
                  "background": "#D3D3D3",
                  "height": 5,
                  "left": 75,
                  "position": "absolute",
                  "width": 200,
                }
              }
            />
            <div
              style={
                Object {
                  "background": "green",
                  "height": 5,
                  "left": 75,
                  "position": "absolute",
                  "width": NaN,
                }
              }
            />
            <br />
          </div>
        </div>
        <br />
        <div
          onClick={[Function]}
        >
          <div>
            <div
              style={
                Object {
                  "position": "absolute",
                }
              }
            >
              3 stars
            </div>
            <div
              style={
                Object {
                  "background": "#D3D3D3",
                  "height": 5,
                  "left": 75,
                  "position": "absolute",
                  "width": 200,
                }
              }
            />
            <div
              style={
                Object {
                  "background": "green",
                  "height": 5,
                  "left": 75,
                  "position": "absolute",
                  "width": NaN,
                }
              }
            />
            <br />
          </div>
        </div>
        <br />
        <div
          onClick={[Function]}
        >
          <div>
            <div
              style={
                Object {
                  "position": "absolute",
                }
              }
            >
              2 stars
            </div>
            <div
              style={
                Object {
                  "background": "#D3D3D3",
                  "height": 5,
                  "left": 75,
                  "position": "absolute",
                  "width": 200,
                }
              }
            />
            <div
              style={
                Object {
                  "background": "green",
                  "height": 5,
                  "left": 75,
                  "position": "absolute",
                  "width": NaN,
                }
              }
            />
            <br />
          </div>
        </div>
        <br />
        <div
          onClick={[Function]}
        >
          <div>
            <div
              style={
                Object {
                  "position": "absolute",
                }
              }
            >
              1 stars
            </div>
            <div
              style={
                Object {
                  "background": "#D3D3D3",
                  "height": 5,
                  "left": 75,
                  "position": "absolute",
                  "width": 200,
                }
              }
            />
            <div
              style={
                Object {
                  "background": "green",
                  "height": 5,
                  "left": 75,
                  "position": "absolute",
                  "width": NaN,
                }
              }
            />
            <br />
          </div>
        </div>
      </div>
      <div>
        <br />
        Size
      </div>
      <div>
        <br />
        Comfort
      </div>
    </div>
    <div
      className="reviews-list"
    >
      <div>
        <div>
          0 reviews, sorted by...
        </div>
        <select
          onChange={[Function]}
          style={
            Object {
              "background": "transparent",
              "border": "none",
              "bottom": 18,
              "left": 150,
              "position": "relative",
            }
          }
        >
          <option
            value="relevant"
          >
            Relevant
          </option>
          <option
            value="helpful"
          >
            Helpful
          </option>
          <option
            value="newest"
          >
            Newest
          </option>
        </select>
      </div>
      <div />
      <button
        className="review-button"
        onClick={[Function]}
        type="button"
      >
        Show More
      </button>
      <button
        className="review-button"
        type="button"
      >
         Add A Review + 
      </button>
    </div>
  </div>,
]
`;
